assets/js/Viewer.js

```diff
@@ -1,17 +1,21 @@
 import React, { Component } from 'react';
+import Sign from './Sign';
 
 class Viewer extends Component {
   render() {
     return(
-      <ul>
-        {this.props.messages.map( msg => {
-          return(
-            <li key={msg}>{msg}</li>
-          );
+      <div>
+        {this.sortedKeys(this.props.signs).map( (key, index) => {
+          let lines = this.props.signs[key];
```

Comment:
let/const are blocked scoped so this can be const since you don't change it

I'd recommend adding eslint with the airbnb config to catch stuff like this, it's a good ES6 teacher. https://www.npmjs.com/package/eslint-config-airbnb